packages/manager/apps/key-management-service/src/pages/onboarding/index.tsx

```diff
@@ -1,13 +1,22 @@
-import React from 'react';
-import { useTranslation } from 'react-i18next';
+import { GuideItem, OnboardingLayout } from '@ovhcloud/manager-components';
+import React, { useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
-import { OnboardingLayout } from '@ovhcloud/manager-components';
-import onboardingImgSrc from './onboarding-img.png';
+import { useTranslation } from 'react-i18next';
+import { useKmsGuides } from '@/hooks/useKmsGuides';
 import { ROUTES_URLS } from '@/routes/routes.constants';
+import onboardingImgSrc from './onboarding-img.png';
 
 export default function Onboarding() {
+  const [onboardingGuide, setOnboardingGuide] = useState<GuideItem>();
   const { t } = useTranslation('key-management-service/onboarding');
   const navigate = useNavigate();
+  const { kmsOnboardingGuide } = useKmsGuides();
+
+  useEffect(() => {
+    if (kmsOnboardingGuide) {
+      setOnboardingGuide(kmsOnboardingGuide);
+    }
+  }, [kmsOnboardingGuide]);
```

Comment:
I don't think you need a state and an useEffect here
And if you need one, it should be in the hook `useKmsGuides` (but you shouldn't since kmsOnboardingGuide is dependant on a state so it should trigger a re-render anyways).

Reply:
fixed